File: app/models/bid.rb
class Bid
  attr_reader :face_value, :number

  def initialize(number, face_value)
    @face_value = face_value
    @number = number
  end

  def >(bid)
    return true if bid.nil?

    if number == bid.number
      face_value > bid.face_value
    else
      number > bid.number
    end
  end

  def to_s
    "#{number} #{face_value}#{'s' unless number == 1}"
  end
end

File: app/assets/javascripts/bid.js
Bid = function (number, faceValue) {
    this.number = number;
    this.faceValue = faceValue;

    this.plusOne = function() {
        return new Bid(this.number + 1, this.faceValue);
    };

    this.toString = function() {
        var ret = this.number + " " + this.faceValue;
        if (this.number != 1)
            ret += "s";
        return ret;
    };

    this.toDice = function() {
        return this.number + ' ' + DICE_ICONS[this.faceValue];
    };

    this.lessThanOrEqual = function(bid) {
        if (this.number < bid.number)
            return true;
        if (this.number > bid.number)
            return false;
        return this.faceValue <= bid.faceValue;
    };

    this.isValid = function() {
        if (this.number < 0)
            // Must bid a postive number
            return false;
        if (this.faceValue < 1 || this.faceValue > 6)
            // faceValue needs to be between 1 and 6
            return false;

        return true;
    };
};
